tests/pytest_tests/unit_tests/test_launch/test_environment/test_aws.py

```diff
@@ -239,3 +239,25 @@ async def _mock_get_session(*args, **kwargs):
     with pytest.raises(LaunchError) as e:
         await environment.upload_file("source_file", "s3a://bucket/key")
         assert e.content == "Destination s3a://bucket/key is not a valid s3 URI."
+
+
+@pytest.mark.asyncio
+async def test_get_partition(mocker):
```

Comment:
We should also have a test where the partition is something other than `aws` since that case is what caused us to need this functionality in the first place